apps/site/lib/site_web/controllers/schedule/line_controller.ex

```diff
@@ -82,11 +83,19 @@ defmodule SiteWeb.ScheduleController.LineController do
         stops: simple_stop_map(conn),
         direction_id: conn.assigns.direction_id,
         route_patterns: conn.assigns.route_patterns,
-        shape_map: conn.assigns.shape_map
+        shape_map: conn.assigns.shape_map,
+        line_diagram: Enum.map(conn.assigns.all_stops, &update_route_stop_data/1)
       }
     )
   end
 
+  def update_route_stop_data({data, %RouteStop{} = map}) do
+    %{
+      stop_data: %{branch: List.first(Keyword.keys(data)), type: List.first(Keyword.values(data))},
```

Comment:
I punted on making this exact because I know it's for the line diagram but that's not this task.

It's a list of tuples like [{:stop, nil}] or [{:line, Braintree}, {:stop, Ashmont}] etc and Poison won't handle encoding tuples, so I made it into a list of maps for JS with presumed keys to identify each tuples intention.